backend/src/v5/services/journaling/index.js

```diff
@@ -0,0 +1,120 @@
+/**
+ *  Copyright (C) 2024 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+const { logInvitationAction, logPermissionAction, logUserAction } = require('../../models/teamspaces.audits');
+const Yup = require('yup');
+const { actions } = require('../../models/teamspaces.audits.constants');
+const { events } = require('../eventsManager/eventsManager.constants');
+const { logger } = require('../../utils/logger');
+const { subscribe } = require('../eventsManager/eventsManager');
+const { types } = require('../../utils/helper/yup');
+
+const userAdded = async ({ teamspace, executor, user }) => {
+	try {
+		await logUserAction(teamspace, actions.USER_ADDED, executor, user);
```

Comment:
Can we let `models/teamspaces/audits` depict the actions instead?

This would be prone to error, as soeone could write a USER_ADDED action with the permissions function.

Reply:
Same with the other functinos you created here